Expensify/react-native-onyx
# Remove metrics

### Details

During the TS migration of metrics, we had a [discussion on Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1708940842987279) if this feature is being used. No one raised any concerns so this PR removes metrics from the codebase (aside from PerformanceUtils which are used in a few other places).

### Related Issues

https://github.com/Expensify/App/issues/34341

### Automated Tests


### Manual Tests


Aside from making sure that onyx works and Performance utils didn't change in functionality there's nothing to test.

To make testing easier just run `npm run build` and paste dist file into expensify project.
1. Open Onyx.js inside node_modules/react-native-onyx
2. Find function init( and after the first conditional paste code below

```
    PerformanceUtils.setShouldDebugSetState(true);
    PerformanceUtils.logSetStateCall({ displayName: 'test', key: 'testKey' }, {}, {});
    PerformanceUtils.logSetStateCall({ displayName: 'test', key: 'testKey' }, { test: 1 }, { test2: 123 });
    PerformanceUtils.logSetStateCall({ displayName: 'test', key: 'testKey' }, { test: 1 }, { test: 123 });
    PerformanceUtils.logSetStateCall({ displayName: 'test', key: 'testKey' }, { test: 1, test2: 1 }, { test2: 123 });
    PerformanceUtils.logSetStateCall({ displayName: 'test', key: 'testKey' }, { test: 1, test2: 1, object: { test: 1, test2: 1 } }, { test2: 123, object: { test: 41, test2: 21 } });
```
3. You can play around with values, check what's printed in the console and compare them with how it used to work before

### Author Checklist


### Screenshots/Videos

N/A
